logger.js

```diff
@@ -28,16 +28,16 @@ function pinoLogger (opts, stream) {
   return loggingMiddleware
 
   function onResFinished (err) {
-    this.removeListener('finish', onResFinished)
     this.removeListener('error', onResFinished)
+    this.removeListener('finish', onResFinished)
 
     var log = this.log
     var responseTime = Date.now() - this[startTime]
 
-    if (err) {
+    if (err || this.err || this.statusCode >= 400) {
```

Comment:
I think this should be `>= 500`, not `400`.

Reply:
I think that you'd want to have logs of issues like: bad request, unauthorized, forbidden access

And besides, we want to log the request / response, and a 4xx is a request (client) error.

Reply:
Non of them are errors.
This is a flow for error conditions.

@jsumners what do you think?

Reply:
One more point on my side, if I'd look at a logs dashboard, I'd be looking to see 4xx response, these can indicate a broken client side library, or a broken flow somewhere.

Reply:
Looks like it got resolved before I came around. I'm fine with the current solution.